### ISSUE TITLE
Cookie Generation - Path

### ISSUE THREAD

**Issue by AlbertoMalavasi** [51Degrees/device-detection-php-onpremise#9]
We have set up an on-premise server in a test domain: 51.change2.io. Its purpose is to provide the JavaScript (js) that needs to be included in a second domain, for instance, test.change2.io. On the on-premise server, we have included the call to the function $flowdata->javascriptbuilder->javascript.

In the second domain, we included the JavaScript using the HTML tag <script src="https://51.change2.io"></script>. When we load the page test.change2.io, the fod.complete() function is never invoked.

However, if we place the HTML page present in the test.change2.io domain in the same path as the file that generates the JavaScript, and therefore inside 51.change2.io, the fod.complete() function is executed correctly.

The same issue occurs if we put the HTML page in a path in a subdirectory; the fod.complete() doesn't work, for example: 51.change2.io/test.

Analyzing the cookies using Chrome's inspector, we notice that the cookie that is created has the path of the domain that contains the HTML file, so for example: test.change2.io. If we change the cookie by setting the domain to 51.change2.io, the fod.complete() function still doesn't work correctly.

In the case of the subdirectory of the same domain, for instance: 51.change2.io/test where we find the cookie created as domain 51.change2.io and with path set as /test. If we proceed to delete the path and set / as the path of the file that generates the js, everything starts working correctly.

**Comment by jwrosewell** [comment]
Move to on-premise as the issue specifically relates to the on-premise deployment.

**Comment by jwrosewell** [comment]
I have created a branch of the main branch called [issue/9](https://github.com/51Degrees/device-detection-php-onpremise/tree/issue/9).

Please can you fork this repo and branch and then add the information necessary under the `examples/onpremise` folder to recreate the issue? Then make a pull request with your changes so that we can clearly see what is needed to recreate the problem.

So far our engineer has copied the files provided across and can't recreate the problem experienced in a development environment. The command used to run the PHP server after the instructions to build the on-premise solution have been followed is.

```
php -c php.ini -S localhost:3000
```

The fod.complete method is resolving for the promise within the web browser.

Adding development SSL certificates and test domains would be helpful.

**Comment by AlbertoMalavasi** [comment]
Thanks for your help. It seems there might be some miscommunication. The branch, issue/9, replicates exactly the state of our main branch. Currently, there is no additional code to add or change to reproduce the problem as it is already present in this state. Have you tried altering the path, for instance, changing the folder? This (https://51.change2.io/) is the link responding to the webpage located at https://51.change2.io/demonew/. If the webpage is located at https://51.change2.io/ it works, if the webpage is located https://51.change2.io/demonew/ it doesn't work.

**Comment by Sara51D** [comment]
Thank you @AlbertoMalavasi. We are looking into this issue and one more piece of information that we need from you is an understanding of how you have setup the service, including the Cross Origin Resource Sharing configuration. If you could please add the configuration of your server to the project, we can then recreate this in a controlled environment and progress further on resolving this issue.

**Comment by AlbertoMalavasi** [comment]
Dear Sara, what configuration did you have that worked when you tested our issue? 
Our current configuration is: Ubuntu 18.04.6 LTS, Apache2 php 7.2.24. CORS: 
```
<IfModule mod_ssl.c>
<VirtualHost *:443>

        ServerAdmin info@change2.it
        DocumentRoot /var/www/html/device-detection-php-onpremise/examples/onpremise/

        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

ServerName 51.change2.io
Include /etc/letsencrypt/options-ssl-apache.conf
SSLCertificateFile /etc/letsencrypt/live/51.change2.io/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/51.change2.io/privkey.pem
</VirtualHost>
</IfModule>
```

**Comment by lucasneting** [comment]
Once again we have tried to recreate a server from scratch but have been unable to overcome the indicated issues. We summarize here the steps taken:

We installed Ubuntu with Ubuntu 18.04.6 LTS, Apache2, and PHP 7.4.
We created a vhost called 51.change2.io.
In the folder var/www/html, we proceeded to download the on-premise project from GitHub.
We then compiled the project following the instructions on GitHub.
We updated the php.ini file to link the .hash file.
We updated the php.ini file as per your instruction to get it to respond correctly.
Inside the folder /var/www/html/device-detection-php-onpremise/examples/onpremise/static page.php, we proceeded to delete all the code and leave only that present from line 224 to 226, thus leaving $output($flowdata->javascriptbuilder->javascript);
We went on a different server and created the test.html page where we include the JavaScript script generated on 51.change2.io via the script tag in the header .
We also inserted in the test.html file the JavaScript code window.onload = function() { fod.complete(function (data) { … The fod.complete will never be executed.
